mesa/examples/basic/boid_flockers/agents.py

```diff
@@ -47,25 +54,49 @@ def __init__(
         self.cohere_factor = cohere
         self.separate_factor = separate
         self.match_factor = match
-        self.neighbors = None
 
     def step(self):
         """Get the Boid's neighbors, compute the new vector, and move accordingly."""
```

Comment:
why these extensive changes in this step method?

Reply:
Mainly performance, a lot of the boid might not have any neighbors at some time, which is where the early exit might add a lot of performance.